pkg/check/datacenter_consistency.go

```
@@ -0,0 +1,86 @@
+				vcConfig.Datacenters,
+				fd.Topology.Datacenter, util.CloudConfigNamespace)
+			errs = append(errs, err)
+			klog.Warningf("%v", err)
+		}
+	}
+
+	for _, fd := range infra.Spec.PlatformSpec.VSphere.FailureDomains {
+		for _, vc := range infra.Spec.PlatformSpec.VSphere.VCenters {
+			if vc.Server == fd.Server {
+				if !containsString(vc.Datacenters, fd.Topology.Datacenter) {
+					err := fmt.Errorf("Datacenter-Consistency: failure domain %q references datacenter %q "+
```

golangci-lint (2.11.4)

[error] ST1005: error strings should not end with punctuation or newlines

(staticcheck)